for_curators/sample_local_dataset_upload_dev.py.ipynb

```diff
@@ -0,0 +1,160 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "92063ed7",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "import os\n",
+    "import requests\n",
+    "import threading\n",
+    "from botocore.credentials import RefreshableCredentials\n",
+    "from botocore.session import get_session\n",
+    "from datetime import datetime, timezone\n",
+    "\n",
+    "\n",
+    "# This script performs the upload of a local datafile to a given Collection (as identified by its Collection uuid),\n",
+    "# where the datafile becomes a Dataset accessible via the Data Portal UI. In order to use this script, you must have\n",
+    "# a Curation API key, you must know the id of the Collection to which you wish to upload the datafile, and you must \n",
+    "# decide upon a string tag (the `curator_tag`) to use to uniquely identify the resultant Dataset within this \n",
+    "# Collection going forward. Uploads to a tag that has not been used yet in the given Collection will result in a\n",
+    "# new Dataset being created. Uploads to a tag for which there already exists a Dataset in the given Collection will \n",
+    "# result in the existing Dataset being replaced by the new Dataset created from the datafile that you are uploading.\n",
+    "\n",
+    "# You can only add/replace Datasets in private Collections or revision Collections. \n",
+    "\n",
+    "\n",
+    "# Curators: substitute field values here ↓\n",
+    "api_key_file = \"api_key_file.txt\"  # path to file containing your API key\n",
+    "filename = \"/absolute/path/to_datafile.h5ad\"  # Provide the absolute path to the h5ad datafile to upload\n",
```

Comment:
I understand the sentiment but I disagree. I want this to be very easy-to-use, and putting the variables that require curator input at the top, where they can plug-and-play, seems to make the most sense.

thoughts on this, @atolopko-czi  @metakuni ?

Reply:
I suppose it comes down to whether you're aiming for this file to take a pedagogical approach--a step-by-step notebook with nice formatting--or be easily executed--an all-in-one script.  If notebook, I'd do what Trent suggests (move vars closer to code); if script, keep 'em at the top.

Reply:
I say notebook... but a _tractable_ notebook. It's way easier for people to put all values to fill in at the top, grouped. If they want to know where they're involved they can go find those spots. Otherwise they have to go looking for value that need to be filled in.

Reply:
I think the issue is that you're introducing things to the reader before they know what it's for, which is an anti-pattern in a notebook. But I realize we're have a 50/50 pedagogical/executable goal for this notebook, so I don't feel strongly about it!

Reply:
> > I'm not a fan of the naming used. How about something like _example/curation_api_ or _python-notebooks/curation_api/_.
> 
> Hmm, now that you mention it, does this notebook even belong in the portal repo? Should it live in https://github.com/chanzuckerberg/single-cell-curation? 🤔
> 
> Since:
> 
> ```
> cellxgene curation tools
> This repository contains documents and code used by cellxgene's curation team. 
> ```

I think this is a great point cc @Bento007